src/script/api/script_goal.hpp

```diff
@@ -89,6 +89,15 @@ class ScriptGoal : public ScriptObject {
 	 */
 	static bool IsValidGoal(GoalID goal_id);
 
+	/**
+	 * Check whether this is a valid goal destination.
+	 * @param company The relevant company if a story page is the destination.
+	 * @param type The type of the goal.
+	 * @param destination The destination of the \a type type.
+	 * @return True if and only if this goal destination is valid.
+	 */
+	static bool IsValidGoalDestination(ScriptCompany::CompanyID company, GoalType type, SQInteger destination);
```

Comment:
These functions should be mentioned in the changelog for GS (`src/script/api/game_changelog.hpp`)